app/src/main/java/pm/gnosis/heimdall/ui/dialogs/share/BaseShareQrCodeDialog.kt

```diff
@@ -51,36 +51,38 @@ abstract class BaseShareQrCodeDialog : BaseDialog() {
         super.onStart()
         disposables += dataSourceObservable()
                 .observeOn(AndroidSchedulers.mainThread())
-                .subscribeBy(onNext = { (name, data) -> onResult(name, data) })
+                .subscribeBy(onNext = { (name, data) -> onResult(name, data) }, onError = Timber::e)
 
         // When clicking "outside" of the dialog we should dismiss it
         disposables += root().clicks()
-                .subscribeBy(onNext = { dismiss() })
+                .subscribeBy(onNext = { dismiss() }, onError = Timber::e)
 
         // Override the root clicks (we don't want to dismiss the dialog)
         disposables += include_base_share_qr_code_background.clicks()
-                .subscribeBy()
+                .subscribeBy(onError = Timber::e)
 
         disposables += generateQrCodeSubject
                 .flatMapSingle {
-                    qrCodeGenerator.generateQrCode(data())
-                            .mapToResult()
-                            .observeOn(AndroidSchedulers.mainThread())
-                            .doOnSubscribe { onQrCodeLoading(true) }
-                            .doOnEvent { _, _ -> onQrCodeLoading(false) }
+                    data()?.let {
+                        qrCodeGenerator.generateQrCode(it)
+                                .mapToResult()
+                                .observeOn(AndroidSchedulers.mainThread())
+                                .doOnSubscribe { onQrCodeLoading(true) }
+                                .doOnEvent { _, _ -> onQrCodeLoading(false) }
```

Comment:
I think here we can use `.doFinally { onQrCodeLoading(false) }` because we don't really care about the success/error values.